packages/ui/src/machines/authenticator/signUp.ts

```diff
@@ -247,28 +241,17 @@ export function createSignUpMachine({ services }: SignUpMachineOptions) {
         setCodeDeliveryDetails,
         setUser,
         sendUpdate: sendUpdate(), // sendUpdate is a HOC
+        setAutoSignInIntent: assign({ intent: (_) => 'autoSignIn' }),
```

Comment:
What's the reasoning for setting the `intent` property to be a function here? I don't see anywhere that we call `intent()` as a function, so maybe I'm misunderstanding something about how the `assign` call is used here.

Reply:
Yeah, it's a function used by `assign` to set the value of `intent` to `"autoSignIn"`. More details on their [docs](https://xstate.js.org/docs/guides/context.html#assign-action).

You see this pattern in https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/machines/authenticator/actions.ts (with @eddiekeller's great comments!)

Reply:
Gotcha, so this specifically gives us reusability for our actions. Makes sense!